Remove timestamp field from telemetry events (#1522)
ai-dynamo/nixl: docs/telemetry.md

### Telemetry Details

> - Events are recorded after the corresponding operation completes. The shared buffer does not attach a per-event timestamp; ordering still reflects enqueue order.

Reviewer: ```suggestion
- Events are recorded after the corresponding operation completes and appear in order.
```

I think it's redundant to document that something is not there, we should focus on the things that exist in the code.
